packages/labs/redux/README.md

```diff
@@ -0,0 +1,231 @@
+
+#### Using the `ContextProvider` controller
+
+```js
+import {ContextProvider, storeContext} from '@lit-labs/redux';
+import {store} from './store.js';
+
+export class AppElement extends LitElement {
+
+  _storeProvider = new ContextProvider(this, {
+    context: storeContext,
+    initialValue: store,
+  });
+
+  ...
+}
+```
+
+#### Using the `@provide()` decorator
+
+```ts
+import {provide, storeContext} from '@lit-labs/redux';
+import {store} from './store.js';
+
+export class AppElement extends LitElement {
+  @provide({context: storeContext})
+  private _store = store;
+
+  ...
+}
+```
+
+### Connect a child component to the provided store
+
+Note: The component must be a child of the provider above. Attempting to connect without a store provided via context will throw an error.
+
+#### Using the `Connector` controller
+
+```js
+import {Connector} from '@lit-labs/redux';
+import {increment} from './counter-slice.js';
+
+export class CounterElement extends LitElement {
+  _connector = new Connector(this, {
+    selector: (state) => state.counter.value,
+  });
+
+  _incrementCount() {
+    this._connector.dispatch(increment());
+  }
+
+  render() {
+    return html`
+      <span>Count: ${this._connector.selected}</span>
+      <button @click=${this.incrementCount}>+</button>
+    `;
+  }
+}
+```
+
+The `Connector` constructor takes an options object as its second argument with the following properties:
+
+**`selector`**
+
+A function that takes state and returns a selected value. A memoized selector may be used like one created with `reselect`.
+
+If no selector function is provided, the controller will not subscribe to Redux store changes nor provide any selected value. Do this if you only wish to bring in the `dispatch` method to the component.
+
+**`equalityCheck`**
+
+A function used to check whether a selected value is different from the previously selected value.
+
+It defaults to triple equals which will suffice for directly selecting values out of the state that's updating with immutable pattern, or if using a memoized selector using a library like [`reselect`](https://github.com/reduxjs/reselect).
+
+Provide a custom function here if the selector returns derived data that's not memoized. The `@lit-labs/redux` package also exports a `shallowEquals` function for convenience.
+
+---
+
+The connector instance provides the following public properties:
+
+**`selected`**
+
+The value selected from state by the selector. It will be `undefined` if no selector was provided in the options.
+
+**`dispatch`**
+
+The `dispatch` method from the Redux store. Use this in other component methods to dispatch actions.
+
+#### Using the `@connect()` and `@dispatch()` decorators
+
+```ts
+import {select, dispatch} from '@lit-labs/redux';
+import {increment} from './counter-slice.js';
+
+export class CounterElement extends LitElement {
+  @select((state) => state.counter.value)
+  _count;
+
+  @dispatch()
```

Comment:
I find this a little odd. Is there a way to vend a utility function instead, so the code below can do:

```ts
  _incrementCount() {
    dispatch(this, increment());
  }
```

Or maybe the decorator could go on the dispatching method?

```ts
  @dispatch()
  _incrementCount() {
    return increment();
  }

```

Reply:
Yes, I did think this was awkward. Kind of figured we can iron out a bit more from user feedback while the package is in labs.

I considered the idea of `dispatch` being provided to a decorated method somehow like your second suggestion, which is similar in idea to [`react-redux`'s `mapDispatchToProps`](https://react-redux.js.org/using-react-redux/connect-mapdispatch#defining-mapdispatchtoprops-as-a-function).